compound_disease/compound_treats_disease/transfer_learning_experiment/data/label_functions/compound_gene_lf.py

```diff
@@ -0,0 +1,560 @@
+
+def LF_HETNET_BINDINGDB(c):
+    """
+    This label function returns 1 if the given Disease Gene pair is
+    located in the BindingDB database
+    """
+    return 1 if (c.Gene_cid, c.Compound_cid, "BindingDB") in knowledge_base else 0
+
+def LF_HETNET_PDSP_KI(c):
+    """
+    This label function returns 1 if the given Disease Gene pair is
+    located in the PDSP_KI database
+    """
+    return 1 if (c.Gene_cid, c.Compound_cid, "PDSP Ki") in knowledge_base else 0
+
+def LF_HETNET_US_PATENT(c):
+    """
+    This label function returns 1 if the given Disease Gene pair is
+    located in the US PATENT database
+    """
+    return 1 if (c.Gene_cid, c.Compound_cid, "US Patent") in knowledge_base else 0
+
+def LF_HETNET_PUBCHEM(c):
+    """
+    This label function returns 1 if the given Disease Gene pair is
+    located in the PUBCHEM database
+    """
+    return 1 if (c.Gene_cid, c.Compound_cid, "PubChem") in knowledge_base else 0
+
+def LF_HETNET_CG_ABSENT(c):
+    """
+    This label function fires -1 if the given Disease Gene pair does not appear 
+    in the databases above.
+    """
+    return 0 if any([
+        LF_HETNET_DRUGBANK(c),
+        LF_HETNET_DRUGCENTRAL(c),
+        LF_HETNET_ChEMBL(c),
+        LF_HETNET_BINDINGDB(c),
+        LF_HETNET_PDSP_KI(c),
+        LF_HETNET_US_PATENT(c),
+        LF_HETNET_PUBCHEM(c)
+    ]) else -1
+
+
+# obtained from ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/ (ncbi's ftp server)
+# https://github.com/dhimmel/entrez-gene/blob/a7362748a34211e5df6f2d185bb3246279760546/download/Homo_sapiens.gene_info.gz <-- use pandas and trim i guess
+columns = [
+    "tax_id", "GeneID", "Symbol",
+    "LocusTag", "Synonyms", "dbXrefs",
+    "chromosome", "map_location", "description",
+    "type_of_gene", "Symbol_from_nomenclature_authority", "Full_name_from_nomenclature_authority",
+    "Nomenclature_status", "Other_designations", "Modification_date"
+]
+gene_desc = pd.read_table("https://github.com/dhimmel/entrez-gene/blob/a7362748a34211e5df6f2d185bb3246279760546/download/Homo_sapiens.gene_info.gz?raw=true", sep="\t", names=columns, compression="gzip", skiprows=1)
+
+
+def LF_CG_CHECK_GENE_TAG(c):
+    """
+    This label function is used for labeling each passed candidate as either pos or neg.
+    Keyword Args:
+    c- the candidate object to be passed in.
+    """
+    sen = c[1].get_parent()
+    gene_name = re.sub("\)", "", c[1].get_span().lower())
+    gene_id = sen.entity_cids[c[1].get_word_start()]
+    gene_entry_df = gene_desc.query("GeneID == @gene_id")
+
+    if gene_entry_df.empty:
+        return -1
+
+    for token in gene_name.split(" "):
+        if gene_entry_df["Symbol"].values[0].lower() == token or token in gene_entry_df["Synonyms"].values[0].lower():
+            return 0
+        elif token in gene_entry_df["description"].values[0].lower():
+            return 0
+    return -1
+
+"""
+SENTENCE PATTERN MATCHING
+"""
+
```

Comment:
Just to clarify, is "binding_indication" the relationship on your edge and the different strings here are the specific pattern you are searching for in the sentence?

Are these patterns generated by you based on reading?  Or is there some source?